universe/ValueRefs.h

```diff
@@ -1695,34 +1633,53 @@ Operation<T>::Operation(OpType op_type, std::vector<std::unique_ptr<ValueRef<T>>
     m_op_type(op_type),
     m_operands(std::move(operands))
 {
-    DetermineIfConstantExpr();
+    InitConstInvariants();
     CacheConstValue();
 }
 
 template <typename T>
-void Operation<T>::DetermineIfConstantExpr()
+void Operation<T>::InitConstInvariants()
 {
     if (m_op_type == RANDOM_UNIFORM || m_op_type == RANDOM_PICK) {
         m_constant_expr = false;
+        this->m_root_candidate_invariant = false;
```

Comment:
Is there anything shadowing `m_root_candidate_invariant` and friends?  If not I don't see a need for `this->`.

Reply:
GCC won't compile without it, in the header-defined cases.

Reply:
Can you be more specific than `won't compile`?

Reply:
https://travis-ci.org/github/freeorion/freeorion/jobs/698338097#L368

Reply:
Oh, I wasn't aware of this behaviour: https://isocpp.org/wiki/faq/templates#nondependent-name-lookup-members